### PR TITLE
chore: add DRPC tailnet & cli implementation for network telemetry

### PR DESCRIPTION
**Pull request by ethanndickson**
- Adds a `TelemetrySink` interface to `tailnet.Conn` to support sending network telemetry events regardless of source.
- Adds a `TelemetrySink` implemention for `tailnetApiConnector`, to be used by the CLI.
- Adds a way to disable network telemetry from the CLI

## TODO:
- [ ] Tests

### CODE REVIEW

**Comment by ethanndickson** [comment]
> [!WARNING]
> <b>This pull request is not mergeable via GitHub because a downstack PR is open. Once all requirements are satisfied, merge this PR as a stack <a href="https://app.graphite.dev/github/pr/coder/coder/13677?utm_source=stack-comment-downstack-mergeability-warning" >on Graphite</a>.</b>
> <a href="https://graphite.dev/docs/merge-pull-requests">Learn more</a>

* **#13677** <a href="https://app.graphite.dev/github/pr/coder/coder/13677?utm_source=stack-comment-icon" target="_blank">Graphite</a> 👈
* **#13675** <a href="https://app.graphite.dev/github/pr/coder/coder/13675?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* `main`

This stack of pull requests is managed by Graphite. <a href="https://stacking.dev/?utm_source=stack-comment">Learn more about stacking.</a>
<h2></h2>

 Join @ethanndickson and the rest of your teammates on <a href="https://graphite.dev?utm-source=stack-comment">Graphite <b>Graphite</b></a>

File: cli/ping.go [hunk 1]
```diff
@@ -58,6 +58,9 @@ func (r *RootCmd) ping() *serpent.Command {
 				_, _ = fmt.Fprintln(inv.Stderr, "Direct connections disabled.")
 				opts.BlockEndpoints = true
 			}
+			if r.noNetworkTelemetry {
+				opts.EnableTelemetry = true
+			}
```
Comment (kylecarbs) [inline cli/ping.go:63]: Something seems off with the naming here.

Comment (ethanndickson) [reply cli/ping.go:63]: I left a comment with the `EnableTelemetry` definition on this, the reasoning being that we default initialize `DialAgentOptions` in tests, where it makes the most sense to have network telemetry disabled.

Comment (kylecarbs) [reply cli/ping.go:63]: I'm just saying the naming doesn't make a lot of sense. If there is no network telemetry, why would we enable telemetry?

Comment (ethanndickson) [reply cli/ping.go:63]: oops

**Pull request closed without merging**